src/pages/permissions/panels/GroupOrIdentityChangesTable.tsx

```diff
@@ -0,0 +1,224 @@
+) => {
+  const identityIDs = Object.keys(identityGroupsChangeSummary);
+
+  const rows: JSX.Element[] = [];
+  for (const id of identityIDs) {
+    const groupChangesForIdentity = identityGroupsChangeSummary[id];
+    const identityLoggedIn = id === loggedInIdentityID;
+    const addedGroups: JSX.Element[] = [];
+    const removedGroups: JSX.Element[] = [];
+
+    for (const group of groupChangesForIdentity.added) {
+      addedGroups.push(
+        <p
+          className="u-no-padding--top modified-item"
+          key={`${id}-${group}-added`}
+        >
+          + {group}
+        </p>,
+      );
+    }
+
+    for (const group of groupChangesForIdentity.removed) {
+      removedGroups.push(
+        <p
+          className="u-no-padding--top modified-item"
+          key={`${id}-${group}-removed`}
+        >
+          - <span className="removed">{group}</span>
+        </p>,
+      );
+    }
+
+    rows.push(
+      <tr key={id} className="modified-row">
+        <td>
+          <>
+            {groupChangesForIdentity.name}
+            <Tag isVisible={identityLoggedIn}>You</Tag>
+          </>
+        </td>
+        <td>{addedGroups.concat(removedGroups)}</td>
+      </tr>,
+    );
+  }
+
+  return rows;
+};
+
+const generateRowsFromGroupIdentityChanges = (
+  groupIdentitiesChangeSummary: ChangeSummary,
+  loggedInIdentityID: string,
+  identities: LxdIdentity[],
+) => {
+  const groups = Object.keys(groupIdentitiesChangeSummary);
+  const identityNameLookup: Record<string, string> = {};
+  identities.forEach(
+    (identity) => (identityNameLookup[identity.id] = identity.name),
+  );
+
+  const rows: JSX.Element[] = [];
+  for (const group of groups) {
+    const identityChangesForGroup = groupIdentitiesChangeSummary[group];
+    const addedIdentities: JSX.Element[] = [];
+    const removedIdentities: JSX.Element[] = [];
+
+    for (const identity of identityChangesForGroup.added) {
+      const identityLoggedIn = identity === loggedInIdentityID;
+      addedIdentities.push(
+        <p
+          className="u-no-padding--top modified-item"
+          key={`${group}-${identity}-added`}
+        >
+          +&nbsp;{identityNameLookup[identity]}
+          <Tag isVisible={identityLoggedIn}>You</Tag>
+        </p>,
+      );
+    }
+
+    for (const identity of identityChangesForGroup.removed) {
+      const identityLoggedIn = identity === loggedInIdentityID;
+      removedIdentities.push(
+        <p
+          className="u-no-padding--top modified-item"
+          key={`${group}-${identity}-removed`}
+        >
+          -&nbsp;<span className="removed">{identityNameLookup[identity]}</span>
+          <Tag isVisible={identityLoggedIn}>You</Tag>
+        </p>,
+      );
+    }
+
+    rows.push(
+      <tr key={group} className="modified-row">
+        <td>{group}</td>
+        <td>{addedIdentities.concat(removedIdentities)}</td>
+      </tr>,
+    );
+  }
+
+  return rows;
+};
+
+const GroupsOrIdentityChangesTable: FC<Props> = ({
```

Comment:
The height of this table is changing when having a long group name in it. Not sure what's causing it?

[Screencast from 02.05.2024 11:36:25.webm](https://github.com/canonical/lxd-ui/assets/1155472/95c3ff6f-022c-4085-a38d-e2b0b8462dcd)

Reply:
the issue was due to cells in the right hand column consists of <p> tags which has bottom margin, but cells in the left hand column does not have <p> tags. So there is height difference for the same text when columns are swapped. This is now resolved by wrapping text in the first column with <p> tags.